src/saga/adaptors/shell/shell_file.py

```diff
@@ -803,6 +803,31 @@ def get_size (self, tgt_in) :
     # ----------------------------------------------------------------
     #
     @SYNC_CALL
+    def exists_self (self):
+
+        self._is_valid ()
+
+        return self.exists (self.url)
+   
+   
+    # ----------------------------------------------------------------
+    #
+    @SYNC_CALL
+    def exists (self, tgt_in):
+
+        self._is_valid ()
+
+        cwdurl = saga.Url (self.url) # deep copy
+        tgt    = saga.Url (tgt_in)   # deep copy
+
+        ret, out, _ = self._command (" test -e '%s'" % tgt.path)
```

Comment:
We always use an sh or ssh channel, and all commands go over that channel.  Only the actual movement of files is done via the sftp/scp layer.  The switch to the file transfer layer is done here: https://github.com/radical-cybertools/saga-python/blob/devel/src/saga/utils/pty_shell.py#L949